### PR TITLE
CDD-2595_Transformation_v2

### PR DESCRIPTION
**Pull request by glidester**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala [hunk 1]
```diff
@@ -41,43 +42,49 @@ class StatusController @Inject()(val validateAndExtractHeadersAction: ValidateAn
...
+            logger.info(s"Declaration information request by ${searchType.getClass.getSimpleName} processed successfully.")(id)
+            logger.debug(s"Returning filtered declaration status request with status code 200 and body\n ${res.body}")(id)
+            Ok(res.body).withConversationId.as(ContentTypes.XML)
+          case Left(errorResult) =>
+            errorResult
+        }
+
+      case _ =>
+        logger.error(s"Not yet available ${searchType.label}: ${searchType.value}")
+        Future.successful(ErrorResponse(NOT_IMPLEMENTED, ErrorResponse.NotImplemented, "Not yet available").XmlResult)
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala]: Should add `withConversationId` so conversation header is returned with error response.

File: app/uk/gov/hmrc/customs/declarations/information/services/StatusResponseFilterService.scala [hunk 2]
```diff
@@ -19,15 +19,66 @@ package uk.gov.hmrc.customs.declarations.information.services
...
+  private def constructInputPrefixToOutputPrefixMap(inputPrefixToUriMap: Map[String, String], outputUriToPrefixMap: Map[String, String]): Map[String, String] = {
+    inputPrefixToUriMap.keySet.foldLeft(Seq.empty[(String, String)]){ (inputPrefixToOutputPrefix, inputPrefix) =>
+      val inputUri = inputPrefixToUriMap(inputPrefix)
+      val outputPrefix = outputUriToPrefixMap.getOrElse(inputUri, inputPrefix)
+      val inputToOutputPrefix: (String, String) = (inputPrefix, outputPrefix)
+
+      inputPrefixToOutputPrefix :+ inputToOutputPrefix
+    }.toMap
+  }
+}
```
Comment (paulblakehmrc) [inline app/uk/gov/hmrc/customs/declarations/information/services/StatusResponseFilterService.scala]: new line

File: app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala (hunk 1 above)
Comment (glidester) [reply app/uk/gov/hmrc/customs/declarations/information/controllers/StatusController.scala]: Ok, added this.

File: app/uk/gov/hmrc/customs/declarations/information/services/StatusResponseFilterService.scala (hunk 2 above)
Comment (glidester) [reply app/uk/gov/hmrc/customs/declarations/information/services/StatusResponseFilterService.scala]: Added

**Pull request merged**